tls/common.go

```diff
@@ -1483,3 +1493,18 @@ func isSupportedSignatureAlgorithm(sigAlg SignatureScheme, supportedSignatureAlg
 	}
 	return false
 }
+
+// CertificateVerificationError is returned when certificate verification fails during the handshake.
```

Comment:
I am wondering whether this means we could simplify how we handle TLS handshake certificate errors.